### DISCUSSION TITLE
IsUUID Decorator problem.

### DISCUSSION THREAD

**Discussion by karenfnet** [typestack/class-validator#2531] [Questions & Answers]
@IsUUID works for Body() in Nest.js but doesn't work for Param() with its DTO, and I have global validation pipe, so everything must be validated to pass to controller, is there any solution?

**Comment by sunnypatell** [comment]
@karenfnet `@Param()` receives raw strings from the URL, so class-validator decorators on a DTO may not work the same way as `@Body()`. for UUID validation on route params, use NestJS's built-in `ParseUUIDPipe` directly:

```ts
@Get(':id')
findOne(@Param('id', new ParseUUIDPipe({ version: '4' })) id: string) {
  return this.service.findOne(id);
}
```

this validates the param before it reaches the controller and returns a 422 if the UUID is invalid.

if you still want to use a DTO approach for params, make sure your `ValidationPipe` has `transform: true` and `transformOptions: { enableImplicitConversion: true }` so the DTO is properly instantiated from the param string.

but for single-param validation, `ParseUUIDPipe` is the recommended pattern.

ref: [NestJS ParseUUIDPipe docs](https://docs.nestjs.com/pipes#built-in-pipes)
